src/Faithlife.WebRequests/WebServiceRequestBase.cs

```diff
@@ -329,6 +336,24 @@ private void SetCookie(HttpResponseHeaders headers, Uri requestUri)
 			}
 		}
 
+		private static WebServiceException WrapRequestException(HttpRequestMessage webRequest, Exception exception)
+		{
+			(exception as WebException).Response?.Dispose();
+			return new WebServiceException(message: "Error building request.", requestMethod: webRequest.Method.Method, requestUri: webRequest.RequestUri, innerException: exception);
+		}
+
+		private static bool ShouldWrapException(Exception exception)
+		{
+			// InvalidDataException can be thrown if GZip stream header returned from server is invalid.
```

Comment:
I wonder if this is true for `HttpClient`, but better safe than sorry?

Reply:
I haven't been able to repro the failure (with `HttpWebRequest` or `HttpClient`) using a simple HTTP server that's designed to return invalid data. I might just keep it, though.